targets/TARGET_ARM_SSG/TARGET_CM3DS_MPS2/gpio_api.c

```diff
@@ -0,0 +1,189 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stddef.h>
+#include "gpio_api.h"
+#include "pinmap.h"
+
+#define GPIO_NUM  4
+
+CMSDK_GPIO_TypeDef* GPIO_MAP[GPIO_NUM] = {
+    CMSDK_GPIO0,
+    CMSDK_GPIO1,
+    CMSDK_GPIO2,
+    CMSDK_GPIO3
+};
+
+#define RESERVED_MISC_PIN 7
+
+/* \brief Gets the FPGA MISC (Miscellaneous control) bit position for the given
+ *        pin name
+ *
+ *  FPGA MISC bit mapping:
+ *    [31:7] Reserved
+ *    [6] CLCD_BL_CTRL
+ *    [5] CLCD_RD
+ *    [4] CLCD_RS
+ *    [3] CLCD_RESET
+ *    [2] Reserved
+ *    [1] SPI_nSS
+ *    [0] CLCD_CS
+ *
+ *  \param[in] pin  MISC pin name
+ *
+ *  \return FPGA MISC bit position
+ */
+static uint8_t get_fpga_misc_pin_pos(PinName pin)
+{
+    uint8_t pin_position = RESERVED_MISC_PIN;
+
+    if (pin == SPI_SCLK) {
+        pin_position = 0;
+    } else if (pin == CLCD_SSEL) {
+        pin_position = 1;
+    } else if (pin == CLCD_RESET) {
+        pin_position = 3;
+    } else if (pin == CLCD_RS) {
+        pin_position = 4;
+    } else if (pin == CLCD_RD) {
+        pin_position = 5;
+    } else if (pin == CLCD_BL_CTRL){
+        pin_position = 6;
+    }
+
+    return pin_position;
+}
+
+uint32_t gpio_set(PinName pin)
+{
+    /* TODO */
```

Comment:
not supported because target does not need to set the pin to gpio or not yet implemented?